contracts/router/quoter/PoolQuoterV1.sol

```diff
@@ -0,0 +1,398 @@
+import {UniversalTokenLib} from "../libs/UniversalToken.sol";
+
+/// @notice Stateless abstraction to calculate exact quotes for any DefaultPool instances.
+abstract contract PoolQuoterV1 is ISwapQuoterV1 {
+    IDefaultPoolCalc internal immutable _defaultPoolCalc;
+    address internal immutable _weth;
+
+    constructor(address defaultPoolCalc, address weth) {
+        _defaultPoolCalc = IDefaultPoolCalc(defaultPoolCalc);
+        _weth = weth;
+    }
+
+    // ═══════════════════════════════════════════ SPECIFIC POOL QUOTES ════════════════════════════════════════════════
+
+    /// @inheritdoc ISwapQuoterV1
+    function calculateAddLiquidity(address pool, uint256[] memory amounts) external view returns (uint256 amountOut) {
+        // Forward the only getter that is not properly implemented in the StableSwap contract (DefaultPool).
+        return _defaultPoolCalc.calculateAddLiquidity(pool, amounts);
+    }
+
+    /// @inheritdoc ISwapQuoterV1
+    function calculateSwap(
+        address pool,
+        uint8 tokenIndexFrom,
+        uint8 tokenIndexTo,
+        uint256 dx
+    ) external view returns (uint256 amountOut) {
+        return IDefaultExtendedPool(pool).calculateSwap(tokenIndexFrom, tokenIndexTo, dx);
+    }
+
+    /// @inheritdoc ISwapQuoterV1
+    function calculateRemoveLiquidity(address pool, uint256 amount)
+        external
+        view
+        returns (uint256[] memory amountsOut)
+    {
+        return IDefaultExtendedPool(pool).calculateRemoveLiquidity(amount);
+    }
+
+    /// @inheritdoc ISwapQuoterV1
+    function calculateWithdrawOneToken(
+        address pool,
+        uint256 tokenAmount,
+        uint8 tokenIndex
+    ) external view returns (uint256 amountOut) {
+        return IDefaultExtendedPool(pool).calculateRemoveLiquidityOneToken(tokenAmount, tokenIndex);
+    }
+
+    // ══════════════════════════════════════════════ POOL GETTERS V1 ══════════════════════════════════════════════════
+
+    /// @inheritdoc ISwapQuoterV1
+    function poolInfo(address pool) external view returns (uint256 numTokens, address lpToken) {
+        numTokens = _numTokens(pool);
+        lpToken = _lpToken(pool);
+    }
+
+    /// @inheritdoc ISwapQuoterV1
+    function poolTokens(address pool) external view returns (PoolToken[] memory tokens) {
+        tokens = _getPoolTokens(pool);
+    }
+
+    // ══════════════════════════════════════════════ POOL INSPECTION ══════════════════════════════════════════════════
+
+    /// @dev Returns the LP token address for the given pool, if it exists. Otherwise, returns address(0).
+    function _lpToken(address pool) internal view returns (address) {
+        // Try getting the LP token address from the pool.
+        try IDefaultExtendedPool(pool).swapStorage() returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256,
+            uint256,
+            uint256,
+            address lpToken
```

Comment:
Interface for saddle stableswap implementation returns [more fields](https://github.com/synapsecns/synapse-contracts/blob/013a7f84a54047bdb83df50370c0b7bec22d85d7/contracts/amm/SwapUtils.sol#L61) than this for `swapStorage`?

Reply:
Saddle interface for reference: https://github.com/saddle-finance/saddle-contract/blob/2cfdd05580efb626d79d597034b2cf5ae89b5626/contracts/interfaces/ISwapV2.sol#L30-L41

Arrays and mappings that are located within the struct are not returned in the autogenerated getter from a public struct variable:
https://docs.soliditylang.org/en/v0.8.17/contracts.html#getter-functions